### ISSUE TITLE
Xmpp access broken by Logitech

### ISSUE THREAD

**Issue by nikshriv** [tuck182/harmony-java-client#27]
As you have probably heard, Logitech decided to kill local network access via Xmpp. Any chance you could change to the local websocket API?

**Comment by tuck182** [comment]
I've been reading about that. I'm not really using this these days (I wrote this to integrate with openHAB, but I've been using Homeseer lately), so I'm unlikely to put the time into doing that, but hopefully someone else will pick that up in some form.

**Comment by bwssytems** [comment]
I might give a shot to update to the websockets api, but that may go away someday as well.

**Comment by wborn** [comment]
We have the same issue with openHAB (https://github.com/openhab/openhab2-addons/issues/4411).  @digitaldan also has some plans to look into this. It may save time if we concentrate our efforts.

**Comment by digitaldan** [comment]
I'm in the process of porting this to websockets, I'm hoping to have something working this weekend.  I will post something as soon as i can.  @tuck182 as I'm writing this, I'm realizing the vast majority of the code is around XMPP processing and will go away. ATM, I'm only planning on using the classes in net/whistlingfish/harmony/config , and a few method signatures (sendButton for example).  I think it would be best to start from scratch with a clean project, do you have any objections?

**Comment by tuck182** [comment]
No, that seems totally reasonable to me.

**Comment by bwssytems** [comment]
I can help test. I have many users clamoring for an update as well.

**Comment by pytz** [comment]
There is a beta firmware now which reenables the XMPP access: [Source](https://community.logitech.com/s/question/0D55A00008D4bZ4SAJ/harmony-hub-firmware-update-fixes-vulnerabilities)

**Comment by wborn** [comment]
Yes [they now commit to supporting the XMPP API for local control](https://community.home-assistant.io/t/logitechs-stance-on-local-apis/85842/18) and don't guarantee other undocumented APIs (Websockets API) remain available.

**Comment by digitaldan** [comment]
uhg, i wish they would make up their mind. I just spent the whole day building a websocket library.  I would acutally prefer this over XMPP, much simpler for sure and easier to support, but it sounds like they are leaving the door open to not supporting it down the road.

**Comment by digitaldan** [comment]
So I have a really basic version working using websockets at https://github.com/digitaldan/harmony-client .  I do like how light weight this is without XMPP.  Since logitech is now going back to XMPP this may not be necessary, but I will keep tinkering with it.   Eventually I would like it to be more compatible with the API in this library and also change the name space back to this project's, or maybe org.openhab.harmonyclient and encourage others to work on it.

**Comment by tuck182** [comment]
I think changing to the openhab namespace is totally reasonable. I just
threw this project together originally to tinker with it. I ended up going
a different direction, but I'm glad it's provided a basis for something
useful.
